File: src/test/testdata/inspections/unsafeIterators/ObjectFloatMap.java
package main;

import com.badlogic.gdx.utils.ObjectFloatMap;

import java.util.Iterator;

class ObjectFloatMapTest {

  void test() {

    ObjectFloatMap<String> ofm = new ObjectFloatMap();

    for (ObjectFloatMap.Entry<String> e: <warning>ofm</warning>) {
      System.out.println(e.key);
    }

    for (ObjectFloatMap.Entry<String> e: new ObjectFloatMap.Entries<String>(ofm)) {
      System.out.println(e.key);
    }

    Iterator i1 = <warning>ofm.keys()</warning>;
    Iterator i2 = <warning>ofm.entries()</warning>;
    ObjectFloatMap.Values i3 = <warning>ofm.values()</warning>;
    Iterator i4 = <warning>ofm.entries()</warning>;

  }

}

File: src/test/testdata/inspections/unsafeIterators/ObjectFloatMap.kt
package main

import com.badlogic.gdx.utils.ObjectFloatMap

@Suppress("UNUSED_VARIABLE", "UNCHECKED_CAST", "USELESS_CAST")
class ObjectFloatMapTest {

  var ofm2: ObjectFloatMap<String>? = null

  fun test() {

    val ofm = ObjectFloatMap<Any>()


    for (e in <warning>ofm</warning>) {
      println(e.key)
    }

    for (e in ObjectFloatMap.Entries<Any>(ofm)) {
      println(e.key)
    }

    val i1 = <warning>ofm.keys()</warning>
    val i2 = <warning>ofm.entries()</warning>
    val i3 = <warning>ofm.values()</warning>
    val i4 = <warning>ofm.entries()</warning>

    <warning>ofm2?.keys()</warning>

    <warning>(ofm2 ?: ObjectFloatMap()).values()</warning>

  }

}

